usaspending_api/download/filestreaming/include_file_description.py

```diff
@@ -0,0 +1,25 @@
+import os
+import zipfile
+
+from usaspending_api.common.retrieve_file_from_uri import RetrieveFileFromUri
+from usaspending_api.download.helpers import write_to_download_log as write_to_log
+
+
+def build_file_description(source_file_template, sources):
+
+    # Read in file description ignoring lines that start with #.
+    with RetrieveFileFromUri(source_file_template).get_file_object() as f:
+        file_description_text = "".join(tuple(l.decode("utf-8") for l in f if not l.startswith(b"#")))
+
+    return file_description_text.format(**{source.source_type: source.file_name for source in sources})
+
+
+def include_file_description_in_zip(file_description_text, destination_file_name, working_dir, zipfile_path):
+    write_to_log(message="Adding file description")
+
+    destination_file_path = os.path.join(working_dir, destination_file_name)
+    with open(destination_file_path, "w") as f:
+        f.write(file_description_text)
+
+    zip_file = zipfile.ZipFile(zipfile_path, 'a', compression=zipfile.ZIP_DEFLATED, allowZip64=True)
```

Comment:
I need to investigate (or be shown) how this works to produce the desired results.

Reply:
1) Write the provided text to the provided destination file.
2) Append the newly written file to the provided zip file.

This logic was basically lifted from the existing code.  It's how the download builds up the zip file.  It just continually appends newly generated csv files.

Reply:
I simply needed to run it myself. It works, of course.

It would be better to consolidate the zip file commands to share the same parameters instead of having to duplicate the same settings across different files.

Reply:
I am cleaning this up a bit.  I'm not happy with it in general.  Also adding unit tests.  GO ME!